packages/babel-plugin-proposal-pipeline-operator/test/fixtures/pipeline-operator/async-with-transform-async/expected.js

```diff
@@ -0,0 +1,39 @@
+var incPromise = x => Promise.resolve(x + 1);
+
+var incFuncPromise = Promise.resolve(x => x + 10);
+
+var double = x => x * 2;
+
+var result =
+/*#__PURE__*/
+function () {
+  var _ref = babelHelpers.asyncToGenerator(function* () {
+    var _;
+
+    return _ = 10, yield (0, incPromise)(_);
+  });
+
+  return function result() {
+    return _ref.apply(this, arguments);
+  };
+}();
+
+var result2 =
+/*#__PURE__*/
+function () {
+  var _ref2 = babelHelpers.asyncToGenerator(function* () {
+    var _ref3, _2;
+
+    return _ref3 = (_2 = 10, yield (0, incPromise)(_2)), double(_ref3);
+  });
+
+  return function result2() {
+    return _ref2.apply(this, arguments);
+  };
+}();
+
+function* foo() {
+  var _3;
+
+  return _3 = 42, (yield incFuncPromise)(_3);
```

Comment:
was this intentional? looks weird, but i guess it could test something, but then exec should reflect that test